interpreter/cling/lib/Interpreter/Interpreter.cpp

```diff
@@ -463,8 +463,8 @@ namespace cling {
     const char* Attr = "";
 #endif
 
-#if defined(__GLIBCXX__)
-    const char* cxa_atexit_is_noexcept = LangOpts.CPlusPlus ? " noexcept" : "";
+#if defined(__GLIBCXX__ || _LIBCPP_VERSION)
```

Comment:
This seems incorrect.  From what I can read in the forum post, the compile-time decision given libc++ is to inject the following declaration at runtime (in https://github.com/root-project/root/pull/12047/files#diff-6e2f3604a761eaec5f5cda89a7dda68ab5a46e4369278267764daff49efaff74R487):
```c++
extern "C" int __cxa_atexit(void (*f)(void*), void*, void*);
```

What seems totally off is that, run-time, GNU libstdc++ headers end up being parsed.  This is not supported, AFAIK.  Therefore, to me the problem instead is most likely in finding header search paths for the standard library.